### PR TITLE
[INFRA] Rework

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: .github/workflows/ci_coverage.yml [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
 env:
-  CMAKE_VERSION: 3.16.9
-  SEQAN3_NO_VERSION_CHECK: 1
   SHARG_NO_VERSION_CHECK: 1
   TZ: Europe/Berlin
 
 defaults:
   run:
-    shell: bash -ex {0}
+    shell: bash -Eexuo pipefail {0}
```
Comment (SGSSGene) [inline .github/workflows/ci_coverage.yml:18]: Very nice! :+1:

File: .github/workflows/ci_macos.yml [hunk 2]
```diff
@@ -1,132 +1,84 @@
...
+    shell: bash -Eeuxo pipefail {0}
 
 jobs:
   build:
     name: ${{ matrix.name }}
     runs-on: macos-12
     timeout-minutes: 120
     strategy:
-      fail-fast: true
+      fail-fast: false
```
Comment (SGSSGene) [inline .github/workflows/ci_macos.yml:26]: I think in the app template we should fail-fast....

File: .github/workflows/ci_macos.yml [hunk 3]
```diff
@@ -1,132 +1,84 @@
...
+    steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
-          path: ${{ env.REPOSITORY_NAME }}
-          fetch-depth: 2
-          submodules: recursive
+          fetch-depth: 1
+          submodules: false
```
Comment (SGSSGene) [inline .github/workflows/ci_macos.yml:55]: Oh yeah!

File: cmake/ccache.cmake [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# --------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2023, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2023, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/raptor/blob/main/LICENSE.md
+# --------------------------------------------------------------------------------------------------
```
Comment (SGSSGene) [inline cmake/ccache.cmake:6]:
```suggestion
# SPDX-FileCopyrightText: 2006-2023, Knut Reinert & Freie Universität Berlin
# SPDX-FileCopyrightText: 2016-2023, Knut Reinert & MPI für molekulare Genetik
# SPDX-License-Identifier: BSD-3-Clause
```

File: test/cli/cli_test.hpp [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
-#include <cstdlib>               // system calls
-#include <filesystem>            // test directory creation
-#include <sstream>               // ostringstream
-#include <string>                // strings
+#include <cstdlib>    // system calls
+#include <filesystem> // test directory creation
+#include <sstream>    // ostringstream
+#include <string>     // strings
 
 #pragma once
```
Comment (SGSSGene) [inline test/cli/cli_test.hpp:8]: Does this even compile? :sweat_smile:  The `#pragma once` should go first.

```
#pragma once

#include <gtest/gtest.h>

#include <cstdlib>    // system calls
#include <filesystem> // test directory creation
#include <sstream>    // ostringstream
#include <string>     // strings
```

File: test/cli/cli_test.hpp [hunk 6]
```diff
@@ -59,9 +57,9 @@ struct cli_test : public ::testing::Test
     {
         // Assemble the directory name.
         ::testing::TestInfo const * const info = ::testing::UnitTest::GetInstance()->current_test_info();
-        std::filesystem::path const test_dir{std::string{OUTPUTDIR} +
-                                             std::string{info->test_case_name()} +
-                                             std::string{"."} +
+        std::filesystem::path const test_dir{std::string{OUTPUTDIR} +              //
+                                             std::string{info->test_case_name()} + //
+                                             std::string{"."} +                    //
```
Comment (SGSSGene) [inline test/cli/cli_test.hpp:62]: Why the "slashes"? Is this clang formatting?

**Review by SGSSGene — CHANGES_REQUESTED** [review]

File: .github/workflows/ci_macos.yml (hunk 2 above)
Comment (eseiler) [reply .github/workflows/ci_macos.yml:26]: But it's annoying when one fails, but you don't know whether the others also fail (compiler-specific)

File: cmake/ccache.cmake (hunk 4 above)
Comment (eseiler) [reply cmake/ccache.cmake:6]: I wanted to do the copyright as separate PR

File: .github/workflows/ci_macos.yml (hunk 2 above)
Comment (eseiler) [reply .github/workflows/ci_macos.yml:26]: If the includes have proper guards, it should still compile. I'll move it up and check the other files. Looks like it was even wrong before this pr :D

File: test/cli/cli_test.hpp (hunk 6 above)
Comment (eseiler) [reply test/cli/cli_test.hpp:62]: Yes, it prevents clang-format from putting everything on two lines. Alternatively, I can wrap it in a `// clang-format off/on`?

Comment (SGSSGene) [reply test/cli/cli_test.hpp:62]: that would be even more annoying xD

**Review by SGSSGene — APPROVED** [review]
Good to go!

**Pull request merged**